web/src/pages/NewTransaction/NavigationButtons/DepositPaymentButton.tsx

```diff
@@ -50,32 +49,19 @@
   const { address } = useAccount();
   const { chain } = useNetwork();
   const ensResult = useEnsAddress({ name: sellerAddress, chainId: 1 });
-  const deadlineTimestamp = useMemo(() => new Date(deadline).getTime(), [deadline]);
-  const timeoutPayment = useMemo(() => (deadlineTimestamp - currentTime) / 1000, [deadlineTimestamp, currentTime]);
+  const deadlineTimestamp = useMemo(() => BigInt(Math.floor(new Date(deadline).getTime() / 1000)), [deadline]);
   const isNativeTransaction = sendingToken?.address === "native";
   const transactionValue = useMemo(
     () => (isNativeTransaction ? parseEther(sendingQuantity) : parseUnits(sendingQuantity, 18)),
     [isNativeTransaction, sendingQuantity]
   );
-  const isDeadlinePassed = currentTime >= deadlineTimestamp;
-
-  useEffect(() => {
-    if (isDeadlinePassed) {
-      navigateAndScrollTop("/new-transaction/deadline");
-    }
-  }, [isDeadlinePassed, navigateAndScrollTop]);
-
-  useEffect(() => {
-    const intervalId = setInterval(() => setCurrentTime(Date.now()), 1000);
-    return () => clearInterval(intervalId);
-  }, []);
 
   useEffect(() => {
     setFinalRecipientAddress(ensResult.data || sellerAddress);
   }, [sellerAddress, ensResult.data]);
 
-  const { data: allowance } = useContractRead({
-    enabled: !isNativeTransaction && !isDeadlinePassed,
+  const { data: allowance, refetch: refetchAllowance } = useContractRead({
+    enabled: !isNativeTransaction,
```

Comment:
this is wrapped within a wrapWithToast() function so it logs an error in the notification popup in case it fails + its just a simple approve with previously validated quantity (the quantity is stored in the context of the NewTransactionContext())